plutus-scb/src/Plutus/SCB/Webserver/Server.hs

```diff
@@ -100,27 +105,70 @@ contractSchema ::
     => ContractInstanceId
     -> Eff effs (ContractSignatureResponse t)
 contractSchema contractId = do
+    ContractInstanceState {csContractDefinition} <- getContractInstanceState @t contractId
+    ContractSignatureResponse <$> exportSchema csContractDefinition
+
+getContractInstanceState ::
+       forall t effs.
+       ( Member (EventLogEffect (ChainEvent t)) effs
+       , Member (Error SCBError) effs
+       )
+    => ContractInstanceId
+    -> Eff effs (ContractInstanceState t)
+getContractInstanceState contractId = do
     latestContractStatus <- runGlobalQuery (Query.latestContractStatus @t)
     case Map.lookup contractId latestContractStatus of
-        Nothing -> throwError $ ContractInstanceNotFound contractId
-        Just ContractInstanceState {csContractDefinition} ->
-            ContractSignatureResponse <$> exportSchema csContractDefinition
+        Nothing    -> throwError $ ContractInstanceNotFound contractId
+        Just value -> pure value
+
+invokeEndpoint ::
+       forall t effs.
+       ( Member (EventLogEffect (ChainEvent t)) effs
+       , Member Log effs
+       , Member (ContractEffect t) effs
+       , Member (Error SCBError) effs
+       , Show t
+       )
+    => EndpointDescription
+    -> JSON.Value
+    -> ContractInstanceId
+    -> Eff effs (ContractSignatureResponse t)
+invokeEndpoint endpointDescription payload contractId = do
+  logInfo $ "Invoking: " <> tshow endpointDescription <> " / " <> tshow payload
+  newState :: [ChainEvent t] <- Instance.callContractEndpoint @t contractId endpointDescription payload
+  logInfo $ "Invocation response: " <> tshow newState
+  contractSchema contractId
 
 parseContractId ::
        (Member (Error SCBError) effs) => Text -> Eff effs ContractInstanceId
-parseContractId t = do
-    let u = UUID.fromText t
-    case u of
-        Nothing   -> throwError $ InvalidUUIDError t
-        Just uuid -> pure $ ContractInstanceId uuid
+parseContractId (UUID.fromText -> Just uuid) =
+         pure $ ContractInstanceId uuid
```

Comment:
still not sure how I feel about view patterns, they're big in Idris but I don't see them adding that much whilst it is one more thing to be confusing

Reply:
@j-mueller has seduced me to the dark side. I cannot go back now.

Reply:
Okay, okay, I've taken it out. :-)